File: backend/Entities/Ticket.cs
namespace backend.Entities;

public class Ticket
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string TicketNumber { get; set; } = string.Empty;
    public string TicketTitle { get; set; } = string.Empty;
    public string TicketDescription { get; set; } = string.Empty;
    public Guid CreatedById { get; set; }
    public User CreatedBy { get; set; } = null!;
    public Guid? AssignedToId { get; set; }
    public User? AssignedTo {get; set; }
    public Guid? TeamId { get; set; }
    public Team? Team {get; set; }
    public Guid CategoryId { get; set; }
    public TicketCategory Category {get; set; } = null!;
    public Guid? SubcategoryId { get; set; }
    public TicketSubCategory? Subcategory { get; set; }
    public Guid StatusId { get; set; }
    public TicketStatus Status { get; set; } = null!;
    public Guid PriorityId { get; set; }
    public TicketPriority Priority { get; set; } = null!;
    public Guid ImpactLevelId { get; set; }
    public ImpactLevel ImpactLevel { get; set; } = null!;
    public Guid UrgencyLevelId { get; set; }
    public UrgencyLevel UrgencyLevel { get; set; } = null!;
    public string Subject { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? FirstResponseAt { get; set; }
    public DateTime? ResolvedAt { get; set; }
    public DateTime? ClosedAt { get; set; }
    public DateTime? SlaDueAt { get; set; }
    public Guid? ResolvedById {get; set; }
    public User? ResolvedBy { get; set; }
    public string? Resolution { get; set; }
    public Guid? ResolutionCategoryId { get; set; }
    public ResolutionCategory? ResolutionCategory { get; set; }
    public bool IsDeleted { get; set; }
    public ICollection<TicketAssignment> Assignments { get; set; } = new List<TicketAssignment>(); 
    public ICollection<TicketComment> Comments { get; set; } = new List<TicketComment>();
    public ICollection<SlaRecord> SlaRecords { get; set; } = new List<SlaRecord>();
}

File: frontend/types/ticket.ts
export interface TicketListDto {
    id : string;
    ticketNumber : string;
    ticketTitle : string;
    statusName : string;
    priortyName : string;
    categoryName : string;
    subcategoryName? : string | null;
    createdByName : string;
    assignedToName? : string | null;
    createdAt : string;
}
import { TicketCommentDto } from "./ticket-comment";
import { TicketAttachmentDto } from './ticket-attachment';
export interface TicketDetailDto {
    id : string;
    ticketNumber : string;
    ticketTitle : string;
    ticketDescription : string;
    subject : string;
    teamId? : string | null;
    teamName? : string | null;
    statusId : string;
    statusName : string;
    priorityId : string;
    priorityName : string;
    categoryId : string;
    categoryName : string;
    subcategoryId? : string | null;
    subcategoryName? : string;
    impactLevelId : string;
    impactLevelName : string;
    urgencyLevelId : string;
    urgencyLevelName : string;
    createdById : string;
    createdByName : string;
    assignedToId? : string | null;
    assignedtoName? : string | null;
    createdAt : string;
    firstResponseAt? : string | null;
    resolvedAt? : string | null;
    closedAt? : string | null;
    comments : TicketCommentDto[];
    attachments : TicketAttachmentDto[];
}
export interface TicketCreateDto {
    ticketTitle : string;
    ticketDescription : string,
    subject : string;
    categoryId : string;
    subcategoryId? : string | null;
    priorityId : string;
    impactLevelId : string;
    urgencyLevelId : string;
}
export interface TicketFilterDto {
    search? : string | null;
    statusId? : string | null;
    categoryId? : string | null;
    assignedToId? : string | null;
    createdById? : string | null;
    urgencyLevelId? : string | null;
    impactLevelId? : string | null;
    createdFrom : string | null;
    createdTo : string | null;
    pageNumber? : number;
    pageSize? : number;
}
export interface TicketResponseDto {
    id : string;
    ticketTitle : string;
    ticketDescription : string;
    subject : string;
    statusName : string;
    impactLevelName : string;
    urgencyLevelName : string;
    priorityName : string;
    categoryName : string;
    createdByName : string;
    assignedToName? : string | null;
    createdAt : string;
}
export interface TicketUpdateDto{
    ticketTitle : string;
    ticketDescription : string;
    subject : string;
    categoryId : string;
    subcategoryId? : string | null;
    priorityId : string;
    impactLevelId : string;
    urgencyLevelId : string;
}
